src/main/java/hudson/plugins/ec2/InstanceInfo.java

```diff
@@ -0,0 +1,39 @@
+package hudson.plugins.ec2;
+
+import com.amazonaws.services.ec2.model.Instance;
+import com.amazonaws.services.ec2.model.KeyPair;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InstanceInfo {
+    private final Instance instance;
+    private KeyPair keypair;
+
+    public InstanceInfo(Instance instance, KeyPair keypair) {
+        this.instance = instance;
+        this.keypair = keypair;
+    }
+
+    public Instance getInstance() {
+        return instance;
+    }
+
+    public void setKeypair(KeyPair keypair) {
+        this.keypair = keypair;
+    }
+
+    public KeyPair getKeypair() {
+        return keypair;
+    }
+
+
+    public static List<InstanceInfo> fromInstances(List<Instance> instances, EC2Cloud cloud) {
+        try {
+            KeyPair keyPair = cloud.resolveKeyPair();
+            return instances.stream().map(obj -> new InstanceInfo(obj, keyPair)).collect(Collectors.toList());
+        } catch (java.io.IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Would be more natural to just throw `IOException` out of this method and let callers deal with it.